libs/application/core/src/lib/formUtils.spec.ts

```diff
@@ -184,3 +186,126 @@ describe('extractRepeaterIndexFromField', () => {
     expect(extractRepeaterIndexFromField(field)).toBe(-1)
   })
 })
+
+describe('mergeAnswers', () => {
+  it('should deep merge objects with new data', () => {
+    expect(
+      mergeAnswers(
+        {
+          periods: [{ startDate: '2021-01-29', endDate: '2021-02-19' }],
+          employer: { isSelfEmployed: 'yes' },
+          payments: { bank: '0000000000', union: 'id', pensionFund: 'id' },
+          applicant: { email: 'mockEmail@island.is', phoneNumber: '9999999' },
+          giveRights: { giveDays: 1, isGivingRights: 'no' },
+          otherParent: 'no',
+          singlePeriod: 'no',
+          requestRights: { requestDays: 18, isRequestingRights: 'no' },
+          firstPeriodStart: 'specificDate',
+          personalAllowance: { useAsMuchAsPossible: 'yes' },
+          confirmLeaveDuration: 'specificDate',
+          usePersonalAllowance: 'yes',
+          usePrivatePensionFund: 'no',
+          personalAllowanceFromSpouse: { useAsMuchAsPossible: 'yes' },
+          usePersonalAllowanceFromSpouse: 'yes',
+        },
+        {
+          periods: [
+            { startDate: '2021-01-29', endDate: '2021-02-25', ratio: '100' },
+            { startDate: '2021-02-20', endDate: '2021-03-15', ratio: '100' },
+          ],
+        },
+      ),
+    ).toStrictEqual({
+      periods: [
+        { startDate: '2021-01-29', endDate: '2021-02-25', ratio: '100' },
+        { startDate: '2021-02-20', endDate: '2021-03-15', ratio: '100' },
+      ],
+      employer: { isSelfEmployed: 'yes' },
+      payments: { bank: '0000000000', union: 'id', pensionFund: 'id' },
+      applicant: { email: 'mockEmail@island.is', phoneNumber: '9999999' },
+      giveRights: { giveDays: 1, isGivingRights: 'no' },
+      otherParent: 'no',
+      singlePeriod: 'no',
+      requestRights: { requestDays: 18, isRequestingRights: 'no' },
+      firstPeriodStart: 'specificDate',
+      personalAllowance: { useAsMuchAsPossible: 'yes' },
+      confirmLeaveDuration: 'specificDate',
+      usePersonalAllowance: 'yes',
+      usePrivatePensionFund: 'no',
+      personalAllowanceFromSpouse: { useAsMuchAsPossible: 'yes' },
+      usePersonalAllowanceFromSpouse: 'yes',
+    })
+  })
+
+  it('should return a merge object', () => {
+    expect(
+      mergeAnswers(
+        { id: 'periodStart', value: '2020-20-20' },
+        { id: 'periodStart', value: '2021-01-01' },
+      ),
+    ).toStrictEqual({ id: 'periodStart', value: '2021-01-01' })
+  })
+
+  it('should return a merge array of object with new answers to overwrite the current answers', () => {
```

Comment:
Just curious about one more merge case that might come up from file uploads, or from deleting a period (we had issues with it originally), perhaps we could add a test for it?:

```
 expect(
      mergeAnswers(
        [
          {
            periods: [{ startDate: '2023-12-12', ratio: 100 }],
          },
        ],
        [
          {
            periods: [], // They deleted the period
          },
        ],
      ),
    ).toStrictEqual([
      {
        periods: [],
      },
    ])
```

Reply:
That's a good point, I will add it! thanks